### DISCUSSION TITLE
Custom bars

### DISCUSSION THREAD

**Discussion by rdong8** [nautechsystems/nautilus_trader#576] [Q&A] [answered]
Is there a way to use custom bars with nautilus? I would like to create a bar with nonstandard fields calculated using external data. Just as an example, say I define a bar with the following rules:
1. Every bar has a high, low, average market cap for the period, and timestamp. There is no open, close or volume.
2. A new bar is created when the ratio of the cumulative volume traded to the market cap reaches some threshold.

**Comment by cjdsellers** [comment, marked as answer]
Hi @rdgh8 thanks for your interest

There is a way to do this, firstly you could:
- Define this custom data type, probably inheriting from `Data` rather than `Bar`.
- Write the logic for aggregating these bars in a class inheriting `Actor`, which you could then use to publish the bars onto the general message bus.
- Strategies or other actors could then subscribe to this custom bar datatype.

You could even take it further by defining your own aggregation type, and writing a custom `BarAggregator` but you probably don't need to for now.

Hope that helps!
